src/test/java/codeu/controller/ProfileServletTest.java

```diff
@@ -11,38 +12,71 @@
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
 import java.io.IOException;
+import java.time.Instant;
+import java.util.UUID;
 
 public class ProfileServletTest {
     private ProfileServlet profileServlet;
     private HttpServletRequest mockRequest;
     private HttpServletResponse mockResponse;
     private RequestDispatcher mockRequestDispatcher;
     private HttpSession mockSession;
+    private UserStore mockUserStore;
+    private User fakeUser;
+    private final String TEST_USERNAME = "test_username";
 
     @Before
     public void setup() throws IOException {
 
         profileServlet = new ProfileServlet();
         mockRequest = Mockito.mock(HttpServletRequest.class);
+        mockSession = Mockito.mock(HttpSession.class);
+
+        setupFakeUser();
+
+        mockUserStore = Mockito.mock(UserStore.class);
+        profileServlet.setUserStore(mockUserStore);
+        Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
         mockResponse = Mockito.mock(HttpServletResponse.class);
         mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+
         Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/profile.jsp"))
                 .thenReturn(mockRequestDispatcher);
+
+
+    }
+
+    //Creating fakeUser
+    private void setupFakeUser() {
+        fakeUser =
+                new User(
+                        UUID.randomUUID(),
+                        "test username",
+                        "$2a$10$eDhncK/4cNH2KE.Y51AWpeL8/5znNBQLuAFlyJpSYNODR/SJQ/Fg6",
+                        Instant.now(), "test description");
     }
 
     @Test
     public void testdoGet() throws IOException, ServletException {
+        Mockito.when(mockSession.getAttribute("user")).thenReturn(TEST_USERNAME);
+        Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+        Mockito.when(mockUserStore.getUser(TEST_USERNAME)).thenReturn(fakeUser);
+        System.out.println("name" + mockUserStore.getUser(TEST_USERNAME));
+
         profileServlet.doGet(mockRequest, mockResponse);
         Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
```

Comment:
We also want to verify in the doGet that we are setting the description param properly.

Reply:
Fixed!